src/AutoRest.CSharp/Mgmt/Decorator/ParentDetection.cs

```diff
@@ -44,6 +44,13 @@ private static IEnumerable<MgmtTypeProvider> GetParent(this Resource resource)
         private static IEnumerable<MgmtTypeProvider> GetParent(this OperationSet resourceOperationSet)
         {
             var parentRequestPath = resourceOperationSet.ParentRequestPath();
+
+            if (parentRequestPath.Equals(resourceOperationSet.GetRequestPath()))
+            {
+                // my parent is myself? Only tenant has this attribute, return empty
+                return Enumerable.Empty<MgmtTypeProvider>();
```

Comment:
Consider to add a test

Reply:
It is very unfortunate that this is very hard to add unit test because this involves quite a few entities from the code model, and building the code model in a unit test is not very practical.
I will talk to Feng to see if we have a test project that generates azure.resourcemanager, we could use that as a test for this.

Reply:
Right. In the future we should focus on the testability. I think some of our codes are too coupled, which makes it hard to do unit testing.